### DISCUSSION TITLE
Swup Support for dynamically loading and transitioning pages

### DISCUSSION THREAD

**Discussion by kylewelsby** [freshframework/fresh#860] [Q&A] [answered]
Looking to find a way to transition pages while keeping part of the page active (e.g. a side bar or music player). 

[Swup](https://swup.js.org/) supports loading the content and swapping it out with the new loading content avoiding the browser page load. 

It also can make use of pre-loading pages, so even faster page loads. 

Would having official support for a feature like this make sense? 

Does anyone have early work-arounds?

**Comment by kylewelsby** [comment]
Also trying [Turbo](https://turbo.hotwired.dev/), it works but with a large issue.  loaded islands are not unloaded, especially event handlers. 

Is there any way to trigger the islands to unmount?

**Comment by marvinhagemeister** [comment, marked as answer]
The modern approach to that is view transitions which has already landed in Chrome and is expected to land in other browsers in the future too. Progress on that can be followed here #1532 .
